File: examples/python/02_lookup_by_address.py
"""Look up a property with field projection.

The `fields` parameter is a comma-separated dotted-path projection. The server
trims the response before sending, which is useful when:

- you're piping the JSON to an LLM (smaller token bill),
- you only need a handful of fields and don't want to deserialize 300+.

Run:
    export ZILLAPI_KEY="zk_..."
    python 02_lookup_by_address.py "350 5th Ave, New York, NY 10118"
"""
import os
import sys

import httpx

KEY = os.environ["ZILLAPI_KEY"]
ADDRESS = sys.argv[1] if len(sys.argv) > 1 else "350 5th Ave, New York, NY 10118"

FIELDS = ",".join([
    "zpid",
    "address",
    "price",
    "zestimate",
    "rentZestimate",
    "bedrooms",
    "bathrooms",
    "livingArea",
    "yearBuilt",
    "homeType",
    "homeStatus",
    "taxAssessedValue",
])

with httpx.Client(
    base_url="https://api.zillapi.com",
    headers={"Authorization": f"Bearer {KEY}"},
    timeout=30,
) as client:
    resp = client.get(
        "/v1/properties/by-address",
        params={"address": ADDRESS, "fields": FIELDS},
    )

resp.raise_for_status()
prop = resp.json()["data"]

for key, value in prop.items():
    print(f"{key:20s} {value}")

File: examples/node/02_lookup_by_address.mjs
// Look up a property with a `fields` projection.
//
// The `fields` parameter trims the response server-side. Useful when:
//   - you're piping the JSON to an LLM (smaller token bill),
//   - you only need a handful of fields and don't want to deserialize 300+.
//
// Run:
//   export ZILLAPI_KEY="zk_..."
//   node 02_lookup_by_address.mjs "350 5th Ave, New York, NY 10118"

const KEY = process.env.ZILLAPI_KEY;
if (!KEY) {
  console.error("Set ZILLAPI_KEY env var: export ZILLAPI_KEY=zk_...");
  process.exit(1);
}

const address = process.argv[2] ?? "350 5th Ave, New York, NY 10118";

const FIELDS = [
  "zpid",
  "address",
  "price",
  "zestimate",
  "rentZestimate",
  "bedrooms",
  "bathrooms",
  "livingArea",
  "yearBuilt",
  "homeType",
  "homeStatus",
  "taxAssessedValue",
].join(",");

const url = new URL("https://api.zillapi.com/v1/properties/by-address");
url.searchParams.set("address", address);
url.searchParams.set("fields", FIELDS);

const resp = await fetch(url, {
  headers: { Authorization: `Bearer ${KEY}` },
});

if (!resp.ok) {
  throw new Error(`HTTP ${resp.status}: ${await resp.text()}`);
}

const { data } = await resp.json();
console.log(JSON.stringify(data, null, 2));
